simulations/simulation_runner.py

```diff
@@ -47,42 +46,44 @@ def run(self):
 
     def step(self):
         """ run one round of the simulation """
+        """ this becomes, who is going to make a message and send to the network """
+        """ rather than what explicit paths happen """
         self.round += 1
-        message_paths = self.msg_gen(self.validator_set)
-        affected_validators = {j for _, j in message_paths}
+        new_messages = self._generate_new_messages()
+        received_messages = self._receive_messages()
+        self._update_safe_estimates(received_messages.keys())
 
-        sent_messages = self._send_messages_along_paths(message_paths)
-        new_messages = self._make_new_messages(affected_validators)
-        self._check_for_new_safety(affected_validators)
-
-        self.plot_tool.update(message_paths, sent_messages, new_messages)
+        self.plot_tool.update(new_messages)
         if self.round % self.report_interval == self.report_interval - 1:
             self.plot_tool.plot()
 
-    def _send_messages_along_paths(self, message_paths):
-        sent_messages = {}
-        for sender, receiver in message_paths:
-            last_message = sender.my_latest_message()
-            if last_message:
-                message = last_message
-            else:
-                message = self.network.get_message_from_validator(sender)
-
-            self.network.propagate_message_to_validator(message, receiver)
-            sent_messages[sender] = message
-
-        return sent_messages
+        self.network.time += 1
```

Comment:
The Simulator updates the network's time. In an syncronous simulator, the process that runs the network would also advance time. The network is dumb in that sense an relies upon the container it sits in for the advancement of time.

Reply:
I think it's fine to make it so something external can update this, but why not add a function like `advance_time` (that accepts some integer) to the network?

I might be missing something, but interacting w/ the network seems clearer to me this way. I was initially confused by the fact that network never seems to move forward in time.

Reply:
The other thought I just had is that this opens the possibility that there are messages that should have been delivered but weren't (aka, messages in the past). 

Does this make sense? Maybe we want to make it so validators don't have to repeatedly call receive, and that messages are delivered to them if the time comes around (this could be enforced in the `advance_time` function). Let me know what you think here.

Reply:
I too think it makes sense to have the simulation runner be a synchronous base line w/ the priority queues adding in partial asynchrony.

Interesting idea on advancing time by an integer amount. You could just add another case to receive, something like:

```
if queue.peek()[0] < self.time:
          queue.pop()
```
Although it does seem like kind of an odd way to model message passing failures. Perhaps just a simple probability that `queue.get()[1]` fails could suffice to model drops.

Reply:
I don't want the network to do the delivering. I want whatever simulator the network is running in to either be an access point for validators to ask for messages OR for the simulator to pop the messages off of the Network and deliver them to validators. In either case, the Network itself doesn't care, the next layer up cares (the same layer that would be advancing the network's time.